### ISSUE TITLE
v0.10.9 launch_gateway() method always starts gateway on the default port

### ISSUE THREAD

**Issue by yakov-g** [bartdag/py4j#406]
I'm trying to run two instances of this small snippet.
```
from py4j.java_gateway import launch_gateway
port = launch_gateway(die_on_exit=True)
print(port)
```
prints: 25333 - default py4j port, gateway is started on port 25333

and so another process, trying to do the same, fails on binding.

It works in v0.10.7 - gateway always started on a random port

**Comment by toroc** [comment]
I'm also seeing this issue with a recent update to v0.10.9. I don't see this issue on v0.10.8.1.

**Comment by bartdag** [comment]
There was a change on the Java side to take into account the port number. This may have caused a regression here. I'll see if I can find anything.

**Comment by bartdag** [comment]
Ok, so this commit introduced the bug: https://github.com/bartdag/py4j/commit/f7b7c8759574367335a4a3ffb8a9c7cacca9f16e

and this commit fixed it: https://github.com/bartdag/py4j/commit/4152353ac142a7c6d177e0d8f5d420d92c846a30

I can probably try to release a v0.10.9.1 during the weekend.

**Comment by yakov-g** [comment]
Thank you!

**Comment by bartdag** [comment]
I just released 0.10.9.1, which should fix this issue.
